plugins/catalog-backend/src/ingestion/processors/ApiDefinitionAtLocationProcessor.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ApiEntity, Entity, LocationSpec } from '@backstage/catalog-model';
+import {
+  LocationProcessor,
+  LocationProcessorDataResult,
+  ProcessorContext,
+} from './types';
+
+const DEFINITION_AT_LOCATION_ANNOTATION = 'backstage.io/definition-at-location';
+
+export class ApiDefinitionAtLocationProcessor implements LocationProcessor {
+  async processEntity(
+    entity: Entity,
+    _location: LocationSpec,
+    context: ProcessorContext,
+  ): Promise<Entity> {
+    if (
+      entity.kind === 'API' &&
+      entity.metadata.annotations &&
+      entity.metadata.annotations[DEFINITION_AT_LOCATION_ANNOTATION]
```

Comment:
I'd flip the conditionals around a bit to eliminate cases we don't care about early on. `if (entity.kind !== 'API') { return entity }`

Saves so many bytes in this text file! :grin: More srsly I feel it's easier to read when there aren't big blocks of indented code.